_shared/local_extensions.py

```diff
@@ -158,6 +160,60 @@ def dedent(self, s):
         """Return `s` dedented."""
         return dedent(s)
 
+    def random_port_number(self):
+        """Return a randomly-generated port number."""
+        if not hasattr(self, "_random_port_numbers"):
+            self._random_port_numbers = [
+                i
+                for i in range(1025, 10000)  # Ports <=1024 are privileged on Linux.
+                if i
+                not in (
```

Comment:
```suggestion
                # Ports <= 1024 are privileged on Linux.
                port for i in range(1025, 10000) if port not in (
```

It might be good to separate the logic from the data here by having a "BAD_PORTS" or "TAKEN_PORTS" list or something similar?

Reply:
:+1: Done: https://github.com/hypothesis/cookiecutters/pull/98/commits/1502746ca513954212ab10c39aa0145554873124